### PR TITLE
[#1240] Remove insertion and deletion of each commit in commits.json

### LINKED ISSUE #1240: Remove insertion and deletion of each commit in commits.json
> > Update: can't really remove the insertions and deletions of each commit as they are required in the backend to calculate the variance.
> 
> Please raise an issue for this, we should aim to resolve this in the near future

_Based on my discussion with @eugenepeh in #1192_ 

Since we have the line contribution info of each file type within a commit, we can remove the insertion and deletion info of each commit in the `commits.json` file. 

The insertion and deletion info can be calculated in the front end instead, by summing up all the insertion and deletion of all filetypes within a commit. More info shown in screenshot below:

Screen Shot 2020-04-17 at 8 13 29 PM

In the meantime, the variance can also be calculated in the front end, as currently the calculation of variance is coupled with the insertion and deletion info of each commit.

### PR DESCRIPTION
**Pull request by 0blivious**
Fixes #1240 
```
We now have the line contribution info of each file type within a
commit, we can remove the insertion and deletion info of each
commit in the commits.json file. Same for
authorFinalContributionMap.

The insertion and deletion info can be calculated in the front
end instead, by summing up all the insertion and deletion of
all filetypes within a commit.

Let's remove the insertion & deletion in commit.json file and
calculate them in front end.
```

### CODE REVIEW

File: src/main/java/reposense/commits/model/AuthorDailyContribution.java [hunk 1]
```diff
@@ -29,36 +24,23 @@ public void setDate(Date date) {
...
-    public int getDeletions() {
-        return deletions;
-    }
-
-    public void setDeletions(int deletions) {
-        this.deletions = deletions;
-    }
-
     /**
      * Adds the {@code commitResult} line contribution count into the {@code Author}'s total line contribution count
```
Comment (jamessspanggg) [inline src/main/java/reposense/commits/model/AuthorDailyContribution.java:28]: Can modify the comment to better suit the changed method.

File: src/main/java/reposense/commits/model/AuthorDailyContribution.java [hunk 2]
```diff
@@ -8,16 +8,11 @@
  * Holds the commits made and line contribution count of an {@code Author} for a single day.
  */
 public class AuthorDailyContribution {
```
Comment (jamessspanggg) [inline src/main/java/reposense/commits/model/AuthorDailyContribution.java:10]: Perhaps this class can be changed to `AuthorDailyCommits`?

File: src/main/java/reposense/commits/model/AuthorDailyContribution.java [hunk 3]
```diff
@@ -8,16 +8,11 @@
  * Holds the commits made and line contribution count of an {@code Author} for a single day.
```
Comment (jamessspanggg) [inline src/main/java/reposense/commits/model/AuthorDailyContribution.java:8]: same goes for this comment, can be modified to better suit the changes made

File: src/main/java/reposense/authorship/model/AuthorshipSummary.java [hunk 4]
```diff
@@ -37,17 +32,11 @@ public AuthorshipSummary(List<FileResult> fileResults, List<Author> authors, Lis
      * {@code filePath} by one.
```
Comment (jamessspanggg) [inline src/main/java/reposense/authorship/model/AuthorshipSummary.java:32]: can modify the comment to remove teh `contribution count of {@code author}`

File: src/main/java/reposense/commits/model/AuthorDailyContribution.java (hunk 2 above)
Comment (0blivious) [reply src/main/java/reposense/commits/model/AuthorDailyContribution.java:10]: That would change too many entries and I feel it is not really necessary.

Actually if you want to change the name I feel a better idea is to keep `AuthorDailyContribution` and change `commitResults` to `commits`

**Comment by 0blivious** [comment]
@jamessspanggg can merge?

**Comment by fzdy1914** [comment]
@0blivious Can help resolve the conflicts. Will merge after the conflicts are resolved.

**Pull request merged**